packages/interpreter/src/components/visitor/elseIfStatement.ts

```diff
@@ -0,0 +1,15 @@
+import { ASTNode } from "bhai-lang-parser";
+
+import InterpreterModule from "../../module/interpreterModule";
+import RuntimeException from "../../exceptions/runtimeException";
+import IfStatement from "./ifStatement";
+
+
+export default class ElseIfStatement extends IfStatement {
+  visitNode(node: ASTNode) {
+    if (!InterpreterModule.getCurrentScope().isIfElseLadderRunning()) {
```

Comment:
This shouldn't be runtime exception. Rather it should be handled in parser itself as `SyntaxError`.

Reply:
Yeah, this looked a little weird to me as well while implementing it, the parser should catch it instead of the interpreter.

Reply:
This is done